packages/a2a-server/src/http/app.ts

```diff
@@ -183,9 +188,35 @@ export async function createApp() {
             .json({ error: `Command not found: ${command}` });
         }
 
-        const result = await commandToExecute.execute(context, args ?? []);
-        logger.info('[CoreAgent] Sending /executeCommand response: ', result);
-        return res.status(200).json(result);
+        if (commandToExecute.executeStream) {
```

Comment:
This code is already getting fairly complex, maybe this should be refactored out to a new method so it reads more clearly?